core/src/main/scala/cats/instances/hash.scala

```diff
@@ -0,0 +1,18 @@
+package cats
+package instances
+
+import cats.functor._
+
+/**
+ * @author Tongfei Chen
+ */
+trait HashInstances {
+  implicit val catsFunctorContravariantForHash: Contravariant[Hash] =
+    new Contravariant[Hash] {
+      /** Derive an `Hash` for `B` given an `Hash[A]` and a function `B => A`.
```

Comment:
Small nitpick, I obsess over details, but comment style isn't compatible with either ScalaDoc or JavaDoc. This title needs a new-line.